File: database/spring-jdbc/src/test/java/com/example/repository/UserRepositoryTest.java
package com.example.repository;

import static org.junit.jupiter.api.Assertions.assertEquals;

import java.util.List;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.jdbc.JdbcTest;
import org.springframework.context.annotation.Import;

import com.example.model.User;

@JdbcTest
@Import(UserRepository.class)
public class UserRepositoryTest {
    @Autowired
    private UserRepository repository;

    @Test
    public void findAll() {
        List<User> result = repository.findAll();
        assertEquals(result.size(), 5);
    }
}

File: rdb/spring-jdbc-kotlin/src/test/kotlin/com/example/UserRepositoryTest.kt
package com.example

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.jdbc.JdbcTest
import org.springframework.context.annotation.Import

@JdbcTest
@Import(UserRepository::class)
class UserRepositoryTest {
    @Autowired
    lateinit var userRepository: UserRepository

    @Test
    fun findAll() {
        val users = userRepository.findAll()
        assertEquals(users.size, 5)
    }
}
